Build failed because Timer.cpp issue on Centos 7

### Description

Build triggers error

### Steps to reproduce

I have all dependencies satisfied. Then, under build directory:

`#cmake3 -Dclient=OFF ..`

```
-- The C compiler identification is GNU 7.3.1
-- The CXX compiler identification is GNU 7.3.1
-- Check for working C compiler: /opt/rh/devtoolset-7/root/usr/bin/cc
-- Check for working C compiler: /opt/rh/devtoolset-7/root/usr/bin/cc - works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /opt/rh/devtoolset-7/root/usr/bin/c++
-- Check for working CXX compiler: /opt/rh/devtoolset-7/root/usr/bin/c++ - works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Performing Test COMPILER_HAS_MBIG_OBJ
-- Performing Test COMPILER_HAS_MBIG_OBJ - Failed
-- ##################################################
-- Mumble version:          1.4.0
-- Architecture:            x64
-- Build type:              Release
-- ##################################################
-- Qt5 component found: Core | Version: 5.9.7
-- Qt5 component found: Network | Version: 5.9.7
-- Qt5 component found: Xml | Version: 5.9.7
-- OpenSSL component found: Crypto | Version: 1.0.2k
-- OpenSSL component found: SSL | Version: 1.0.2k
-- Protobuf found | Version: 2.5.0
-- Qt5 component found: Sql | Version: 5.9.7
-- File: /opt/sources/mumble/mumble-1.4.230.src/build/src/murmur/murmur_autogen
-- Qt5 component found: DBus | Version: 5.9.7
-- Ice component found: Ice | Version: 3.7.7
-- Ice component found: IceSSL | Version: 3.7.7
-- Ice optional component found: IceDiscovery | Version: 3.7.7
-- Ice optional component found: IceLocatorDiscovery | Version: 3.7.7
-- Configuring done
-- Generating done
-- Build files have been written to: /opt/sources/mumble/mumble-1.4.230.src/build
```

then `make`

```
Scanning dependencies of target shared_autogen
[  1%] Running cpp protocol buffer compiler on /opt/sources/mumble/mumble-1.4.230.src/src/Mumble.proto
[  3%] Automatic MOC for target shared
[  3%] Built target shared_autogen
Scanning dependencies of target shared
[  5%] Building CXX object src/CMakeFiles/shared.dir/shared_autogen/mocs_compilation.cpp.o
[  6%] Building CXX object src/CMakeFiles/shared.dir/Mumble.pb.cc.o
[  8%] Building CXX object src/CMakeFiles/shared.dir/Ban.cpp.o
[ 10%] Building CXX object src/CMakeFiles/shared.dir/EnvUtils.cpp.o
[ 12%] Building CXX object src/CMakeFiles/shared.dir/FFDHE.cpp.o
[ 13%] Building CXX object src/CMakeFiles/shared.dir/HostAddress.cpp.o
[ 15%] Building CXX object src/CMakeFiles/shared.dir/HTMLFilter.cpp.o
[ 17%] Building CXX object src/CMakeFiles/shared.dir/License.cpp.o
[ 18%] Building CXX object src/CMakeFiles/shared.dir/LogEmitter.cpp.o
[ 20%] Building CXX object src/CMakeFiles/shared.dir/OSInfo.cpp.o
[ 22%] Building CXX object src/CMakeFiles/shared.dir/PasswordGenerator.cpp.o
[ 24%] Building CXX object src/CMakeFiles/shared.dir/PlatformCheck.cpp.o
[ 25%] Building CXX object src/CMakeFiles/shared.dir/QtUtils.cpp.o
[ 27%] Building CXX object src/CMakeFiles/shared.dir/ProcessResolver.cpp.o
[ 29%] Building CXX object src/CMakeFiles/shared.dir/SelfSignedCertificate.cpp.o
[ 31%] Building CXX object src/CMakeFiles/shared.dir/ServerAddress.cpp.o
[ 32%] Building CXX object src/CMakeFiles/shared.dir/ServerResolver.cpp.o
[ 34%] Building CXX object src/CMakeFiles/shared.dir/ServerResolverRecord.cpp.o
[ 36%] Building CXX object src/CMakeFiles/shared.dir/SSL.cpp.o
[ 37%] Building CXX object src/CMakeFiles/shared.dir/SSLCipherInfo.cpp.o
[ 39%] Building CXX object src/CMakeFiles/shared.dir/SSLLocks.cpp.o
[ 41%] Building CXX object src/CMakeFiles/shared.dir/Timer.cpp.o
/opt/sources/mumble/mumble-1.4.230.src/src/Timer.cpp: In static member function ‘static quint64 Timer::now()’:
/opt/sources/mumble/mumble-1.4.230.src/src/Timer.cpp:104:20: error: ‘CLOCK_MONOTONIC’ was not declared in this scope
  if (clock_gettime(CLOCK_MONOTONIC, &ts) != 0) {
                    ^~~~~~~~~~~~~~~
/opt/sources/mumble/mumble-1.4.230.src/src/Timer.cpp:104:6: error: ‘clock_gettime’ was not declared in this scope
  if (clock_gettime(CLOCK_MONOTONIC, &ts) != 0) {
      ^~~~~~~~~~~~~
/opt/sources/mumble/mumble-1.4.230.src/src/Timer.cpp:104:6: note: suggested alternative: ‘clock_t’
  if (clock_gettime(CLOCK_MONOTONIC, &ts) != 0) {
      ^~~~~~~~~~~~~
      clock_t
make[2]: *** [src/CMakeFiles/shared.dir/build.make:364: src/CMakeFiles/shared.dir/Timer.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:136: src/CMakeFiles/shared.dir/all] Error 2
make: *** [Makefile:150: all] Error 2
```

### Mumble version

1.4.230

### Mumble component

Server

### OS

Linux

### Reproducible?

Yes

### Additional information

Investigating about `CLOCK_MONOTONIC` and I have found a  [Redhat documentation about this](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux_for_real_time/7/html/reference_guide/sect-posix_clocks), in example 15.2 and I see there is a header:

`#include <time.h>`

but in Mumble code at `src/Timer.cpp`, line 99 the header is:

`#include <sys/time.h>`

I changed it to `#include <time.h>` and build perfect!

```
Scanning dependencies of target shared_autogen
[  1%] Running cpp protocol buffer compiler on /opt/sources/mumble/mumble-1.4.230.src/src/Mumble.proto
[  3%] Automatic MOC for target shared
[  3%] Built target shared_autogen
Scanning dependencies of target shared
[  5%] Building CXX object src/CMakeFiles/shared.dir/shared_autogen/mocs_compilation.cpp.o
[  6%] Building CXX object src/CMakeFiles/shared.dir/Mumble.pb.cc.o
[  8%] Building CXX object src/CMakeFiles/shared.dir/Ban.cpp.o
[ 10%] Building CXX object src/CMakeFiles/shared.dir/EnvUtils.cpp.o
[ 12%] Building CXX object src/CMakeFiles/shared.dir/FFDHE.cpp.o
[ 13%] Building CXX object src/CMakeFiles/shared.dir/HostAddress.cpp.o
[ 15%] Building CXX object src/CMakeFiles/shared.dir/HTMLFilter.cpp.o
[ 17%] Building CXX object src/CMakeFiles/shared.dir/License.cpp.o
[ 18%] Building CXX object src/CMakeFiles/shared.dir/LogEmitter.cpp.o
[ 20%] Building CXX object src/CMakeFiles/shared.dir/OSInfo.cpp.o
[ 22%] Building CXX object src/CMakeFiles/shared.dir/PasswordGenerator.cpp.o
[ 24%] Building CXX object src/CMakeFiles/shared.dir/PlatformCheck.cpp.o
[ 25%] Building CXX object src/CMakeFiles/shared.dir/QtUtils.cpp.o
[ 27%] Building CXX object src/CMakeFiles/shared.dir/ProcessResolver.cpp.o
[ 29%] Building CXX object src/CMakeFiles/shared.dir/SelfSignedCertificate.cpp.o
[ 31%] Building CXX object src/CMakeFiles/shared.dir/ServerAddress.cpp.o
[ 32%] Building CXX object src/CMakeFiles/shared.dir/ServerResolver.cpp.o
[ 34%] Building CXX object src/CMakeFiles/shared.dir/ServerResolverRecord.cpp.o
[ 36%] Building CXX object src/CMakeFiles/shared.dir/SSL.cpp.o
[ 37%] Building CXX object src/CMakeFiles/shared.dir/SSLCipherInfo.cpp.o
[ 39%] Building CXX object src/CMakeFiles/shared.dir/SSLLocks.cpp.o
[ 41%] Building CXX object src/CMakeFiles/shared.dir/Timer.cpp.o
[ 43%] Building CXX object src/CMakeFiles/shared.dir/UnresolvedServerAddress.cpp.o
[ 44%] Building CXX object src/CMakeFiles/shared.dir/User.cpp.o
[ 46%] Building CXX object src/CMakeFiles/shared.dir/Version.cpp.o
[ 48%] Building CXX object src/CMakeFiles/shared.dir/crypto/CryptographicHash.cpp.o
[ 50%] Building CXX object src/CMakeFiles/shared.dir/crypto/CryptographicRandom.cpp.o
[ 51%] Building CXX object src/CMakeFiles/shared.dir/crypto/CryptStateOCB2.cpp.o
[ 53%] Building CXX object src/CMakeFiles/shared.dir/__/3rdparty/arc4random/arc4random_uniform.cpp.o
[ 55%] Linking CXX static library libshared.a
[ 56%] Built target shared
Scanning dependencies of target mumble-server_autogen
[ 58%] Automatic MOC for target mumble-server
[ 58%] Built target mumble-server_autogen
[ 60%] Generating Ice files
Scanning dependencies of target mumble-server
[ 62%] Building CXX object src/murmur/CMakeFiles/mumble-server.dir/mumble-server_autogen/mocs_compilation.cpp.o
[ 63%] Building CXX object src/murmur/CMakeFiles/mumble-server.dir/main.cpp.o
[ 65%] Building CXX object src/murmur/CMakeFiles/mumble-server.dir/Cert.cpp.o
[ 67%] Building CXX object src/murmur/CMakeFiles/mumble-server.dir/Messages.cpp.o
[ 68%] Building CXX object src/murmur/CMakeFiles/mumble-server.dir/Meta.cpp.o
[ 70%] Building CXX object src/murmur/CMakeFiles/mumble-server.dir/PBKDF2.cpp.o
[ 72%] Building CXX object src/murmur/CMakeFiles/mumble-server.dir/Register.cpp.o
[ 74%] Building CXX object src/murmur/CMakeFiles/mumble-server.dir/RPC.cpp.o
[ 75%] Building CXX object src/murmur/CMakeFiles/mumble-server.dir/Server.cpp.o
[ 77%] Building CXX object src/murmur/CMakeFiles/mumble-server.dir/ServerDB.cpp.o
[ 79%] Building CXX object src/murmur/CMakeFiles/mumble-server.dir/ServerUser.cpp.o
[ 81%] Building CXX object src/murmur/CMakeFiles/mumble-server.dir/__/ACL.cpp.o
[ 82%] Building CXX object src/murmur/CMakeFiles/mumble-server.dir/__/Channel.cpp.o
[ 84%] Building CXX object src/murmur/CMakeFiles/mumble-server.dir/__/ChannelListenerManager.cpp.o
[ 86%] Building CXX object src/murmur/CMakeFiles/mumble-server.dir/__/Connection.cpp.o
[ 87%] Building CXX object src/murmur/CMakeFiles/mumble-server.dir/__/Group.cpp.o
[ 89%] Building CXX object src/murmur/CMakeFiles/mumble-server.dir/UnixMurmur.cpp.o
[ 91%] Building CXX object src/murmur/CMakeFiles/mumble-server.dir/Zeroconf.cpp.o
[ 93%] Building CXX object src/murmur/CMakeFiles/mumble-server.dir/__/__/3rdparty/qqbonjour/BonjourServiceRegister.cpp.o
[ 94%] Building CXX object src/murmur/CMakeFiles/mumble-server.dir/DBus.cpp.o
[ 96%] Building CXX object src/murmur/CMakeFiles/mumble-server.dir/MurmurIce.cpp.o
[ 98%] Building CXX object src/murmur/CMakeFiles/mumble-server.dir/Murmur.cpp.o
[100%] Linking CXX executable ../../mumble-server
[100%] Built target mumble-server
```



### Relevant log output

_No response_

### Screenshots

_No response_

@NormandoHall  sorry to bother you , I have been trying to get murmur to compile on a centos 7.9 VM before i try to update our old wow-mumble , it keeps failing COMPILER_HAS_MBIG_OBJ , but so does yours. So have you seen the following problem?

Like cmake3 crashing somewhere with mentioning FindPythonInterpreter

I am very novice with these things so if you had this issue also because of some sort of missing dependency any help is appreciated :)!

```
cmake3 -Dclient=OFF -Ice=OFF ..
-- The C compiler identification is GNU 4.8.5
-- The CXX compiler identification is GNU 4.8.5
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc - works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ - works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Performing Test COMPILER_HAS_MBIG_OBJ
-- Performing Test COMPILER_HAS_MBIG_OBJ - Failed
-- ##################################################
-- Mumble version:          1.4.0
-- Architecture:            x64
-- Build type:              Release
-- ##################################################
-- Qt5 component found: Core | Version: 5.9.7
-- Qt5 component found: Network | Version: 5.9.7
-- Qt5 component found: Xml | Version: 5.9.7
-- OpenSSL component found: Crypto | Version: 1.0.2k
-- OpenSSL component found: SSL | Version: 1.0.2k
-- Protobuf found | Version: 2.5.0
CMake Error at cmake/qt-utils.cmake:6 (include):
  include could not find load file:

    FindPythonInterpreter
Call Stack (most recent call first):
  src/murmur/CMakeLists.txt:16 (include)


-- Qt5 component found: Sql | Version: 5.9.7
-- File: /home/ward/Documents/githubmumble/mumble/build/src/murmur/murmur_autogen
-- Qt5 component found: DBus | Version: 5.9.7
-- Ice component found: Ice | Version: 3.7.7
-- Ice component found: IceSSL | Version: 3.7.7
-- Ice optional component found: IceDiscovery | Version: 3.7.7
-- Ice optional component found: IceLocatorDiscovery | Version: 3.7.7
-- Configuring incomplete, errors occurred!
See also "/home/ward/Documents/githubmumble/mumble/build/CMakeFiles/CMakeOutput.log".
See also "/home/ward/Documents/githubmumble/mumble/build/CMakeFiles/CMakeError.log".
```
Saw you using devtoolset-7 so tried that also

```
[root@localhost build]# scl enable devtoolset-7 bash
[root@localhost build]# rm  -rf ./*
[root@localhost build]# cmake3 -Dclient=OFF -Ice=OFF ..
-- The C compiler identification is GNU 7.3.1
-- The CXX compiler identification is GNU 7.3.1
-- Check for working C compiler: /opt/rh/devtoolset-7/root/usr/bin/cc
-- Check for working C compiler: /opt/rh/devtoolset-7/root/usr/bin/cc - works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /opt/rh/devtoolset-7/root/usr/bin/c++
-- Check for working CXX compiler: /opt/rh/devtoolset-7/root/usr/bin/c++ - works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Performing Test COMPILER_HAS_MBIG_OBJ
-- Performing Test COMPILER_HAS_MBIG_OBJ - Failed
-- ##################################################
-- Mumble version:          1.4.0
-- Architecture:            x64
-- Build type:              Release
-- ##################################################
-- Qt5 component found: Core | Version: 5.9.7
-- Qt5 component found: Network | Version: 5.9.7
-- Qt5 component found: Xml | Version: 5.9.7
-- OpenSSL component found: Crypto | Version: 1.0.2k
-- OpenSSL component found: SSL | Version: 1.0.2k
-- Protobuf found | Version: 2.5.0
CMake Error at cmake/qt-utils.cmake:6 (include):
  include could not find load file:

    FindPythonInterpreter
Call Stack (most recent call first):
  src/murmur/CMakeLists.txt:16 (include)


-- Qt5 component found: Sql | Version: 5.9.7
-- File: /home/ward/Documents/githubmumble/mumble/build/src/murmur/murmur_autogen
-- Qt5 component found: DBus | Version: 5.9.7
-- Ice component found: Ice | Version: 3.7.7
-- Ice component found: IceSSL | Version: 3.7.7
-- Ice optional component found: IceDiscovery | Version: 3.7.7
-- Ice optional component found: IceLocatorDiscovery | Version: 3.7.7
-- Configuring incomplete, errors occurred!
See also "/home/ward/Documents/githubmumble/mumble/build/CMakeFiles/CMakeOutput.log".
See also "/home/ward/Documents/githubmumble/mumble/build/CMakeFiles/CMakeError.log".
```


The CmakeError.log but I assume this is not actually the reason for it failing.

```
Performing C++ SOURCE FILE Test COMPILER_HAS_MBIG_OBJ failed with the following output:
Change Dir: /home/ward/Documents/githubmumble/mumble/build/CMakeFiles/CMakeTmp

Run Build Command(s):/opt/rh/devtoolset-7/root/usr/bin/gmake cmTC_a0e51/fast && /opt/rh/devtoolset-7/root/usr/bin/gmake  -f CMakeFiles/cmTC_a0e51.dir/build.make CMakeFiles/cmTC_a0e51.dir/build
gmake[1]: Entering directory '/home/ward/Documents/githubmumble/mumble/build/CMakeFiles/CMakeTmp'
Building CXX object CMakeFiles/cmTC_a0e51.dir/src.cxx.o
/opt/rh/devtoolset-7/root/usr/bin/c++    -DCOMPILER_HAS_MBIG_OBJ -fPIE   Wa,-mbig-obj -std=gnu++14 -o CMakeFiles/cmTC_a0e51.dir/src.cxx.o -c /home/ward/Documents/githubmumble/mumble/build/CMakeFiles/CMakeTmp/src.cxx
c++: error: Wa,-mbig-obj: No such file or directory
gmake[1]: *** [CMakeFiles/cmTC_a0e51.dir/build.make:86: CMakeFiles/cmTC_a0e51.dir/src.cxx.o] Error 1
gmake[1]: Leaving directory '/home/ward/Documents/githubmumble/mumble/build/CMakeFiles/CMakeTmp'
gmake: *** [Makefile:141: cmTC_a0e51/fast] Error 2


Source file was:
int main() { return 0; }
Performing C SOURCE FILE Test CMAKE_HAVE_LIBC_PTHREAD failed with the following output:
Change Dir: /home/ward/Documents/githubmumble/mumble/build/CMakeFiles/CMakeTmp

Run Build Command(s):/opt/rh/devtoolset-7/root/usr/bin/gmake cmTC_b912b/fast && /opt/rh/devtoolset-7/root/usr/bin/gmake  -f CMakeFiles/cmTC_b912b.dir/build.make CMakeFiles/cmTC_b912b.dir/build
gmake[1]: Entering directory '/home/ward/Documents/githubmumble/mumble/build/CMakeFiles/CMakeTmp'
Building C object CMakeFiles/cmTC_b912b.dir/src.c.o
/opt/rh/devtoolset-7/root/usr/bin/cc   -DCMAKE_HAVE_LIBC_PTHREAD -fPIE   -o CMakeFiles/cmTC_b912b.dir/src.c.o   -c /home/ward/Documents/githubmumble/mumble/build/CMakeFiles/CMakeTmp/src.c
Linking C executable cmTC_b912b
/usr/bin/cmake3 -E cmake_link_script CMakeFiles/cmTC_b912b.dir/link.txt --verbose=1
/opt/rh/devtoolset-7/root/usr/bin/cc  -DCMAKE_HAVE_LIBC_PTHREAD    CMakeFiles/cmTC_b912b.dir/src.c.o  -o cmTC_b912b 
CMakeFiles/cmTC_b912b.dir/src.c.o: In function `main':
src.c:(.text+0x2f): undefined reference to `pthread_create'
src.c:(.text+0x3b): undefined reference to `pthread_detach'
src.c:(.text+0x47): undefined reference to `pthread_cancel'
src.c:(.text+0x58): undefined reference to `pthread_join'
src.c:(.text+0x6c): undefined reference to `pthread_atfork'
collect2: error: ld returned 1 exit status
gmake[1]: *** [CMakeFiles/cmTC_b912b.dir/build.make:107: cmTC_b912b] Error 1
gmake[1]: Leaving directory '/home/ward/Documents/githubmumble/mumble/build/CMakeFiles/CMakeTmp'
gmake: *** [Makefile:141: cmTC_b912b/fast] Error 2


Source file was:
#include <pthread.h>

void* test_func(void* data)
{
  return data;
}

int main(void)
{
  pthread_t thread;
  pthread_create(&thread, NULL, test_func, NULL);
  pthread_detach(thread);
  pthread_cancel(thread);
  pthread_join(thread, NULL);
  pthread_atfork(NULL, NULL, NULL);
  pthread_exit(NULL);

  return 0;
}

Determining if the function pthread_create exists in the pthreads failed with the following output:
Change Dir: /home/ward/Documents/githubmumble/mumble/build/CMakeFiles/CMakeTmp

Run Build Command(s):/opt/rh/devtoolset-7/root/usr/bin/gmake cmTC_0484b/fast && /opt/rh/devtoolset-7/root/usr/bin/gmake  -f CMakeFiles/cmTC_0484b.dir/build.make CMakeFiles/cmTC_0484b.dir/build
gmake[1]: Entering directory '/home/ward/Documents/githubmumble/mumble/build/CMakeFiles/CMakeTmp'
Building C object CMakeFiles/cmTC_0484b.dir/CheckFunctionExists.c.o
/opt/rh/devtoolset-7/root/usr/bin/cc   -DCHECK_FUNCTION_EXISTS=pthread_create -fPIE   -o CMakeFiles/cmTC_0484b.dir/CheckFunctionExists.c.o   -c /usr/share/cmake3/Modules/CheckFunctionExists.c
Linking C executable cmTC_0484b
/usr/bin/cmake3 -E cmake_link_script CMakeFiles/cmTC_0484b.dir/link.txt --verbose=1
/opt/rh/devtoolset-7/root/usr/bin/cc  -DCHECK_FUNCTION_EXISTS=pthread_create    CMakeFiles/cmTC_0484b.dir/CheckFunctionExists.c.o  -o cmTC_0484b  -lpthreads 
/opt/rh/devtoolset-7/root/usr/libexec/gcc/x86_64-redhat-linux/7/ld: cannot find -lpthreads
collect2: error: ld returned 1 exit status
gmake[1]: *** [CMakeFiles/cmTC_0484b.dir/build.make:107: cmTC_0484b] Error 1
gmake[1]: Leaving directory '/home/ward/Documents/githubmumble/mumble/build/CMakeFiles/CMakeTmp'
gmake: *** [Makefile:141: cmTC_0484b/fast] Error 2
```

--- a/src/Timer.cpp
+++ b/src/Timer.cpp
@@ -96,7 +96,7 @@ quint64 Timer::now() {
 #elif defined(Q_OS_UNIX)
 #	include <errno.h>
 #	include <string.h>
-#	include <sys/time.h>
+#	include <time.h>
 #	include <unistd.h>
 #	if defined(_POSIX_TIMERS) && defined(_POSIX_MONOTONIC_CLOCK)
 quint64 Timer::now() {

